### PR TITLE
[ConvDiff] Elemental unit tests

### PR DESCRIPTION
**Pull request by rubenzorrilla**
Aiming to clean up and prepare the ConvectionDiffusionApplication to be core app, I've created some unit tests of the "referene" elements. In subsequent PRs we can safely clean up and enhance the implementation by using these tests.

### CODE REVIEW

File: applications/ConvectionDiffusionApplication/tests/cpp_tests/test_eulerian_conv_diff.cpp [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        }
+
+        // Test element
+        auto p_element = r_test_model_part.pGetElement(1);
+        Vector RHS = ZeroVector(4);
+        Matrix LHS = ZeroMatrix(4,4);
+        p_element->CalculateLocalSystem(LHS, RHS, r_test_model_part.GetProcessInfo());
+
+        std::vector<double> expected_RHS = {0.0, 0.0, 0.0, 0.0};
+        std::vector<double> expected_LHS = {1.49976, 0.494317, -0.232728, 0.494317,
```
Comment (philbucher) [inline applications/ConvectionDiffusionApplication/tests/cpp_tests/test_eulerian_conv_diff.cpp]: have you seen [this](https://github.com/KratosMultiphysics/Kratos/blob/master/kratos/includes/checks.h#L49-L88)? :D

**Comment by jcotela** [comment]
From my side, :+1: for the tests, but please use the new matrix macros if possible.

**Comment by rubenzorrilla** [comment]
I already updated it to use the new macros.

**Comment by jcotela** [comment]
@rubenzorrilla my bad, I did not see the last commits... Reviewing now.

**Review by jcotela — APPROVED** [review]
:+1:

**Pull request merged**